load.go

```diff
@@ -447,3 +453,57 @@ func (b *ArtifactBuild) processBuildArgs(lex *shell.Lex, args map[string]string,
 
 	return goerrors.Join(errs...)
 }
+
+func (img *ImageConfig) processBuildArgs(lex *shell.Lex, args map[string]string, allowArg func(string) bool) error {
+	var errs error
+	for _, p := range []*string{&img.Base, &img.Cmd, &img.User, &img.Entrypoint, &img.StopSignal, &img.WorkingDir} {
```

Comment:
Likewise, I don't see why image base would need build args.

Reply:
At the very least I think it would be useful for the tag. Some repos do not have a `latest` tag, and not allowing a build arg here forces a hard-coded tag. I'm not married to the idea of introducing this, but it does seem useful.

Other uses could be not hard-coding a registry.

Reply:
Not hard-coding a registry could be taken care of by source policies.

Reply:
What do you think about tags?

Reply:
I think I'd opt for explicitness in the spec in this case until someone actually asks for this.

Reply:
Also base image is now deprecated in favor of the new `bases`.